core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -259,6 +262,7 @@ public void setTransportSupplier(final Supplier<ClientTransport> supplier) {
               }
             }
         });
+        listener.transportInUse(false);
```

Comment:
Should we delay this until after all the `createRealStream()`s, otherwise there is a race. I guess there is a race anyway since there will be some time between creating a stream and the transport notifying the channel that it is in use. Meh.